File: src/lib-tests/test_bind_key.cpp
#include <gtest/gtest.h>
#include <unplayplay.hpp>

using namespace std::string_view_literals;

TEST(BindKey, Unit) {
    auto decrypted = unplayplay::decrypt_key(unplayplay::util::Key("01"sv));
    auto file_id = unplayplay::util::FileId("01"sv);

    auto bind_result = unplayplay::bind_key(decrypted, file_id);
    auto expected = unplayplay::util::Key("E32698DCAFABA7529A223C8E34EC08CC"sv);
    EXPECT_EQ(bind_result, expected);
}

File: src/python-tests/test_bind_key.py
from re_unplayplay import decrypt_key, bind_key


def test_bind_key() -> None:
    decrypted = decrypt_key(bytes.fromhex('01'))
    assert bind_key(decrypted, '01').hex() == 'E32698DCAFABA7529A223C8E34EC08CC'
